test/test_files/common/todatecast.test

```diff
@@ -0,0 +1,34 @@
+-GROUP ToDateCast
+-DATASET CSV empty
+
+--
+
+-CASE ToDateCastTest
+
+-STATEMENT RETURN CAST(date('2024-04-05'), 'date')
+---- 1
+2024-04-05
+
+-STATEMENT RETURN CAST(timestamp('2024-04-05'), 'date')
+---- 1
+2024-04-05
+
+-STATEMENT RETURN CAST(CAST('2024-04-05', 'TIMESTAMP_MS'), 'date')
```

Comment:
What will happen if I cast a `timestamp` with hour, minute and second to `date`?

Reply:
It gives the date part of the timestamp without the hour:minute:second. For example,
`RETURN CAST(timestamp('2024-04-05 23:59:59.999'), 'date')`
gives
`2024-04-05`